File: 2.Implement/Delivery_Server/src/main/java/com/delivery/app/Delivery/data/dto/response/review/GetMyReviewsResponseDto.java
package com.delivery.app.Delivery.data.dto.response.review;

import com.delivery.app.Delivery.data.dto.response.BaseResponseDto;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.ArrayList;

@Getter
public class GetMyReviewsResponseDto implements BaseResponseDto {
    private ArrayList<Long> id;
    private ArrayList<Long> marketId;
    private ArrayList<String> marketName;
    private ArrayList<String> comment;
    private ArrayList<Float> score;
    private ArrayList<LocalDateTime> date;

    public GetMyReviewsResponseDto() {
        id = new ArrayList<>();
        marketId = new ArrayList<>();
        marketName = new ArrayList<>();
        comment = new ArrayList<>();
        score = new ArrayList<>();
        date = new ArrayList<>();
    }
    public boolean isEmpty() {
        return id.isEmpty();
    }
    public void addElement(Long id, Long marketId, String marketName, String comment, Float score, LocalDateTime date) {
        this.id.add(id);
        this.marketId.add(marketId);
        this.marketName.add(marketName);
        this.comment.add(comment);
        this.score.add(score);
        this.date.add(date);
    }
}

File: 2.Implement/DeliveryApp/app/src/main/java/com/project/deliveryapp/retrofit/dto/response/review/GetMyReviewsResponseDto.kt
package com.project.deliveryapp.retrofit.dto.response.review

import com.project.deliveryapp.retrofit.dto.ResponseDto
import com.project.deliveryapp.toList
import org.json.JSONObject
import java.time.LocalDateTime

class GetMyReviewsResponseDto(json: JSONObject): ResponseDto {
    val idList: List<Long>
    val marketIdList: List<Long>
    val marketNameList: List<String>
    val commentList: List<String>
    val scoreList: List<Float>
    val dateList: List<LocalDateTime>

    init {
        val obj = json.getJSONObject("responseDto")

        idList = obj.getJSONArray("id").toList()
        marketIdList = obj.getJSONArray("marketId").toList()
        marketNameList = obj.getJSONArray("marketName").toList()
        commentList = obj.getJSONArray("comment").toList()
        scoreList = obj.getJSONArray("score").toList()
        dateList = obj.getJSONArray("date").toList()
    }
}